### ISSUE TITLE
Should yoda conditionals be a blocker?

### ISSUE THREAD

**Issue by emrikol** [Automattic/VIP-Coding-Standards#3]
Stopping a deploy only because of some missed yoda conditionals is, imho, a bit strong.  Is that something that we want to do and move in to our documentation?

cc @jacklenox

**Comment by david-binda** [comment]
What I really love on yoda conditionals is that they prevent bugs when an assignment is done instead of evaluation. Eg:

```
if ( $a = 1 ) {
```

Is totally valid PHP code, but very likely does something different than what developer meant - the developer meant `if ( $a == 1 ) {`. Enforcing yoda conditionals prevents those kinds of tricky bugs, since the following code throws a Fatal error:

```
if ( 1 = $a ) {
```

That's the theory. The fact is that our PHPCS checks are not flagging assignments inside the conditional, while it's what we are really after.

Thus, in my humble opinion:

1) we should tread yoda as warning
2) we should be flagging assignments inside conditionals as warnings and always check with developer before deploying such code

**Comment by mattoperry** [comment]
I think that assignments inside conditionals are almost always a mistake.  I'd be okay if those were blockers really ... even if they were intended they're probably wrong.

Yoda itself is a source of controversy.  Some people think (correctly I think) that it makes code less readable, and should therefore be avoided.  [WordPress coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions) says it's okay for equality checks but should be avoided for other comparisons.

In any case, I agree w/ the above -- flag assignments in conditionals, not yoda.

**Comment by jacklenox** [comment]
Sorry @emrikol, I somehow missed this when you first posted it. I agree with the above. I'm not sure I've ever blocked a deploy because of a Yoda conditional, but I maybe have if there were a lot and if they were alongside other significant deviations from the coding standards.
